internal/handlers/boot_artifacts.go

```diff
@@ -0,0 +1,99 @@
+package handlers
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"regexp"
+
+	"github.com/openshift/assisted-image-service/pkg/imagestore"
+	"github.com/openshift/assisted-image-service/pkg/isoeditor"
+	log "github.com/sirupsen/logrus"
+)
+
+type BootArtifactsHandler struct {
+	ImageStore imagestore.ImageStore
+}
+
+var _ http.Handler = &BootArtifactsHandler{}
+
+var bootpathRegexp = regexp.MustCompile(`^/boot-artifacts/(.+)`)
+
+func parseArtifact(path string) (string, error) {
+	match := bootpathRegexp.FindStringSubmatch(path)
+	if len(match) < 1 {
+		return "", fmt.Errorf("malformed download path: %s", path)
+	}
+
+	var artifact string
+	switch match[1] {
+	case "rootfs":
+		artifact = "rootfs.img"
+	case "kernel":
+		artifact = "vmlinuz"
+	default:
+		return "", fmt.Errorf("malformed download path: %s", path)
+	}
+	return artifact, nil
+}
+
+func (b *BootArtifactsHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
```

Comment:
metrics? events?

Reply:
Yeah, we should add this to the metrics. Not sure what you mean by events though.

Reply:
Hm. I think we have events on generating an iso. So is it applicable here as well?